citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -1113,6 +1114,34 @@ protected void doPostJsonResponse(ParameterParser params,
 		return results;
 	}
 
+	private void validateNestedCollection(List<CitationCollectionOrder> citationCollectionOrders, String citationCollectionId) throws Exception {
```

Comment:
It's much better to have a static model of the allowed containers and then a generic validator that check that the object matches the state.

It's really easy to get typo/copy-paste errors.